src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int ARRAY_INDEX_FOR_DATE = 0;
+    private static final int ARRAY_INDEX_FOR_NAME = 1;
+    private static final int ARRAY_INDEX_FOR_WORKING_HOURS = 2;
+    private static final int ARRAY_INDEX_FOR_INCOME_PER_HOUR = 3;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int salary = 0;
+        StringBuilder salaryInfo = new StringBuilder();
+        for (String name : names) {
+            if (name != null) {
```

Comment:
if you add this condition to next if() block where you check equals you will be able to move all code one tabulation to the left and it will look better